include/pybind11/detail/common.h

```diff
@@ -25,8 +25,10 @@
 // on the main `pybind11` namespace.
 #if !defined(PYBIND11_NAMESPACE)
 #    ifdef __GNUG__
-#        define PYBIND11_NAMESPACE pybind11 __attribute__((visibility("hidden")))
+#        define PYBIND11_NS_VISIBILITY(...) __VA_ARGS__ __attribute__((visibility("hidden")))
```

Comment:
nit While `NS` is a good abbreviation, perhaps it is still good to full expand `NAMESPACE`?

Or if this is agnostic to the type of symbol, consider just removing the `NS` qualifier?